### PR TITLE
Dashboard Page Not Updating

### PR DESCRIPTION
**Pull request by bspidel**
The Failed Jobs list on the dashboard page is not loading recent entries

Issue 617

@richmahn Please review. This code works against dev-tx-job but needs refactored to fit your tx-manager paradigm.

### CODE REVIEW

File: libraries/manager/manager.py [hunk 1]
```diff
@@ -552,8 +554,8 @@ def generate_dashboard(self, max_failures=MAX_FAILURES):
                 'html.parser'))
 
             # build job failures table
-
-            job_failures = self.get_job_failures(registered_jobs)
+            job_failures = self.get_failures()
```
Comment (PhotoNomad0) [inline libraries/manager/manager.py]: Need to pass max_failures as parameter to get_job_failures() so count is not hard coded to 10.

File: libraries/manager/manager.py [hunk 2]
```diff
@@ -605,6 +608,16 @@ def generate_dashboard(self, max_failures=MAX_FAILURES):
...
 
+    def get_failures(self):
+        client = boto3.client('dynamodb')
+        return client.query( TableName=u'dev-tx-job',
+            IndexName = u'status-index',
+            Select     = u'ALL_PROJECTED_ATTRIBUTES',
+            KeyConditionExpression=u'#stts = :st',
+            ExpressionAttributeValues = {":st": {"S": "failed"}},
+            ExpressionAttributeNames = {"#stts": "status"},
+            Limit=10)
```
Comment (PhotoNomad0) [inline libraries/manager/manager.py]: Need to change 10 to a parameter of getFailures() such as max_count

File: libraries/manager/manager.py [hunk 3]
```diff
@@ -605,6 +608,16 @@ def generate_dashboard(self, max_failures=MAX_FAILURES):
 
         return dashboard
 
+    def get_failures(self):
+        client = boto3.client('dynamodb')
+        return client.query( TableName=u'dev-tx-job',
```
Comment (PhotoNomad0) [inline libraries/manager/manager.py]: This should not be hard coded to 'dev-tx-job'.  Use self.job_table_name.

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]

File: libraries/manager/manager.py [hunk 4]
```diff
@@ -605,6 +608,16 @@ def generate_dashboard(self, max_failures=MAX_FAILURES):
...
+            IndexName = u'status-index',
+            Select     = u'ALL_PROJECTED_ATTRIBUTES',
+            KeyConditionExpression=u'#stts = :st',
+            ExpressionAttributeValues = {":st": {"S": "failed"}},
+            ExpressionAttributeNames = {"#stts": "status"},
```
Comment (PhotoNomad0) [inline libraries/manager/manager.py]: @bspidel If you change 'status-index' to have secondary sort key on created_at, and then add to the query ScanIndexForward = False, then the most recent failures will be read first.

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]

**Review by PhotoNomad0 — APPROVED** [review]

**Comment by PhotoNomad0** [comment]
@bspidel - Did some cleanup on test cases in PR to add language report to Dashboard.  These are now in develop.

**Comment by richmahn** [comment]
Not needed any more.

**Pull request closed without merging**